dist/save/index.js

```diff
@@ -0,0 +1,134 @@
+// GCS Cache Post Action - Save caches
+
+// *** Imports ***
+// Node.js dependencies
+const fs = require('fs');
+const path = require('path');
+const md5 = require('md5');
+// GitHub Actions dependencies
+const core = require('@actions/core');
+const exec = require('@actions/exec');
+// Google Cloud client library
+// https://github.com/googleapis/nodejs-storage
+const {Storage} = require('@google-cloud/storage');
+
+// *** Configuration ***
+const uploadDirectory = 'temp';
+const compressionCommand = 'tar -zcf';
+const expandCommand = 'tar -zxf';
+const compressedFileExtension = '.tar.gz';
+const removeCommand = 'rm -r';
+
+// TODO: Move commonly used functions into a module
+// const myModule = require('../module');
+// let val = myModule.hello(); // val is "Hello"
+// console.log(val);
+
+async function run() {
+  try {
+    // Read inputs
+    const noDownload = core.getInput('no-download', { required: false });
+    const cacheKey = core.getInput('cache-key', { required: true });
+    const pathsString = core.getInput('paths', { required: true });
+
+    // Check if provided and read environment variables
+    if (!process.env.IOS_GCS_KEYFILE_BASE64) {
+      throw new Error('No GCS keyfile provided as environment variable');
+    }
+    const gcsKeyfileString = process.env.IOS_GCS_KEYFILE_BASE64;
+    // TODO: Check if it is possible to get the paths of the missing caches from the Main Action via environment variable
+    // const missingCachesPaths = process.env.NOT_FOUND_CACHES;
+
+    // Extract GCS Keyfile
+    const buffer = new Buffer.from(gcsKeyfileString, 'base64');
+    const gcsKeyfile = JSON.parse(buffer.toString('utf8'));
+
+    // Create a GCS client
+    const storage = await new Storage({ credentials: gcsKeyfile });
+    const bucketName = gcsKeyfile.project_id;
+    const bucket = await storage.bucket(bucketName);
+
+    // Get all caches available in Bucket for the given cache key
+    const [availableCaches] = await bucket.getFiles({
+      prefix: cacheKey
+    });
+
+    // List all given path with their corresponding cache references (if exists in the bucket)
+    const cacheObjects = pathsString.split('\n').map((cachePath) => {
+      const cacheFileName = fileNameForCache(cacheKey, cachePath);
+      const cacheReference = availableCaches.find((cache) => {
+        return cache.name == cacheFileName;
+      });
+
+      return [cachePath, cacheReference];
```

Comment:
It would be more neat if you returned an object here instead of an array. this way we can use deconstruction and dont have to do the strange array index accessing like `return cachePath[1] === undefined;`

```suggestion
      return {cachePath, cacheReference};
```